### PR TITLE
use icons from icon theme

### PR DESCRIPTION
**Pull request by tgilgien**
This commit makes the app try to use the icons from the current icon theme, and falls back to the bundled ones if none is available.

Icon recoloring is not supported, so themes relying on it (e.g. breeze dark) will look broken (dark icons on dark background). Note that the current implementation always use dark icons even when using a dark color scheme.

### CODE REVIEW

**Comment by Bzero** [comment]
Thanks a lot for this pull request! Using the icon theme does make sense and hopefully makes the application look a bit more native for users that prefer that. 

Could you please run `ruff check` and make sure that your changes adhere to the ruff linting and formatting rules? (I'm sorry that there is no github action in place yet that automatically enforces this, I will add that shorty).

**Review by Bzero — CHANGES_REQUESTED** [review]

File: typstwriter/editor.py [hunk 1]
```diff
@@ -333,7 +333,10 @@ def show_error(self, msg):
         self.gridLayout = QtWidgets.QGridLayout()
 
         self.label_w = QtWidgets.QLabel()
-        self.label_w.setPixmap(QtGui.QPixmap(util.icon_path("warning.svg")))
+        if(QtGui.QIcon.hasThemeIcon(QtGui.QIcon.DialogWarning)):
+            self.label_w.setPixmap(QtGui.QIcon.fromTheme(QtGui.QIcon.DialogWarning).pixmap(64))
```
[the remark this answers is not in the document]
Comment (tgilgien) [reply typstwriter/editor.py]: Resolved, also changed in the if statement

**Comment by tgilgien** [comment]
Thanks for the feedback!

I made the necessary changes to pass ``ruff check``. (Btw, I'm rather new to python development, so don't hesitate to be harsh on style)

**Comment by Bzero** [comment]
Great, this is all looking good to me.

> I made the necessary changes to pass `ruff check`. (Btw, I'm rather new to python development, so don't hesitate to be harsh on style)

I think a quick `ruff format` will fix the remaining formatting issues. I am myself not very picky about code style or formatting but I think following a standard eases the collaboration when multiple people work on a project.

**Comment by Bzero** [comment]
Thanks a lot for your work, very much appreciated!

**Pull request merged**